test/js/util/mapbox.test.js

```diff
@@ -65,6 +65,21 @@ test("mapbox", function(t) {
         t.end();
     });
 
+    t.test('.normalizeSpriteURL', function(t) {
+        t.test('correctly transforms mapbox:// URLs', function(t) {
+            t.equal(
+                mapbox.normalizeSpriteURL('mapbox://mapbox.streets-v8', undefined, '.json'),
+                'https://a.tiles.mapbox.com/styles/v1/mapbox/mapbox.streets-v8/sprite.json@1x?access_token=key'
```

Comment:
Yes, that is how the sprite API was designed. Personally, I don't mind it. cc @jakepruitt @tmcw for background on why it was designed this way.